.github/workflows/build.yml

```diff
@@ -82,7 +82,7 @@ jobs:
       fail-fast: false
       matrix:
         java: [ 17, 11 ]
-    if: ${{ inputs.github-actions-opt-in }}
+    if: ${{ inputs.github-actions-opt-in == 'true' }}
```

Comment:
It seems #17027 works well. The `JUnit tests` run successfully.

Reply:
The issue we're having is that _all_ PRs are running the JUnit tests. Turns out this expression is checking `if: 'true'` or `if: 'false'` which both evaluate to true since they're non-empty strings.

Reply:
thanks for explanation!